lib/bolt_server/transport_app.rb

```diff
@@ -630,6 +634,28 @@ def make_winrm_target(target_hash)
       [400, e.message]
     end
 
+    # Returns a list of targets parsed from a Project inventory
+    #
+    # @param project_ref [String] the project_ref to compute the inventory from
+    get '/project_inventory_targets' do
+      bolt_config = config_from_project(params['project_ref'])
+      if bolt_config.inventoryfile && bolt_config.project.inventory_file.to_s != bolt_config.inventoryfile
+        raise Bolt::ValidationError, "Project inventory must be defined in #{bolt_config.project.inventory_file}"
+      end
+      plugin = Bolt::Plugin.setup(bolt_config, nil)
+      inventory = Bolt::Inventory.from_config(bolt_config, plugin)
+      target_list = inventory.get_targets('all').map do |target|
+        target_data = {
+          'name' => target.name,
+          'transport' => target.transport
+        }
+        target_data.merge(target.to_h)
+      end
+      target_list.to_json
+    rescue Bolt::Error => e
```

Comment:
I thought the opposite here, that we would want to broadly catch any error bolt might throw loading inventory.

Reply:
You're right, we should. I'm just not sure if those errors should return a `400` status because `400 Bad Request` usually means that something's up with the request. Here, the request is fine, just something went wrong w/ the parsing.

We had a similar discussion about this re: what status to return for an unsupported transport on the Orch-side, and we agreed with 422 there. Feels like this should also return a 422.

Reply:
Might be worth bringing up ^ with the team though because an "invalidly structured inventory file" could warrant a different status code than a "valid Bolt inventory file but an invalid Orch inventory file" (which is what led to Orch's 422).

Reply:
I think 500 for everything makes sense for now. We might later want to split it into 500 vs. 501 depending on whether the inventory file is using features that bolt-server doesn't support (I think that would just be plugins?).